st2client/st2client/base.py

```diff
@@ -124,7 +124,7 @@ def get_client(self, args, debug=False):
         password = credentials.get('password', None)
         cache_token = rc_config.get('cli', {}).get('cache_token', False)
 
-        if credentials:
+        if username:
```

Comment:
Now that I think more about it - this will probably work for v2.2.0, but we need to add some more robustness and logic to handle more scenarios.

Good example of scenario which is not handled correctly right now is a case when only a `username` is provided in the config and there is no cached token. In such scenario we don't want to perform an authentication with the auth API since password is mandatory (but that's what happens right now).